### DISCUSSION TITLE
--files-from doesn't work with any --delete options

### DISCUSSION THREAD

**Discussion by trunghn295** [RsyncProject/rsync#407] [General]
When I use the --files-from options to specify the list of files that I want to sync, then everything works fine.
But, I also want to delete the files that are not in the list of sync files, so I tried to specify --delete, --del, --delete-after... but it doesn't work, the files are not in the list of sync files are still there in the target.
I found a question in StackOverflow posted from 2009, but the accepted solution to use --include-from is not what I want, and is not suitable for my situation https://stackoverflow.com/questions/1813907/rsync-delete-files-from-list-dest-does-not-delete-unwanted-files

Could anyone give me some hints?

**Comment by WayneD** [comment]
The manpage discusses how deletions only happen in synchronized directories:
https://download.samba.org/pub/rsync/rsync.1#opt--delete

If you're specifying all the files in a destination directory hierarchy, you should be using includes & excludes. See the [support/files-to-excludes](https://github.com/WayneD/rsync/blob/master/support/files-to-excludes) script for one possible way to turn a list of files into include/exclude rules (and combine with `--delete-excluded` to delete anything else in the transferred hierarchy).

Keep in mind: always run a trial rsync command with `--dry-run` before the real thing to make sure that your new copy is going to do what you expect.
